test/test-viz.cpp

```diff
@@ -0,0 +1,350 @@
+            rect.h = height;
+            rect.w = width;
+            count++;
+            rects.emplace_back(rect);
+
+            for(auto input : elem.ins) {
+                Line connection;
+                connection.x2 = rect.x;
+                connection.y2 = rect.y + rect.h/2;
+                connection.x1 = input.layer * 2 * width + width;
+                connection.y1 = 0;
+                connections.emplace_back(connection);
+            }
+        }
+    }
+
+    else {
+        int count = 0;
+        int freeSpaces = elems.size() + 1;
+        double step = (SCREEN_HEIGHT - height * elems.size()) / freeSpaces;
+        for(auto elem : elems) {
+            SDL_FRect rect;
+            rect.x = elem.layer * 2 * width;
+            rect.y = step * (count + 1) + height * count;
+            rect.h = height;
+            rect.w = width;
+            count++;
+            rects.emplace_back(rect);
+
+            for(auto input : elem.ins) {
+                Line connection;
+                connection.x2 = rect.x;
+                connection.y2 = rect.y + rect.h/2;
+                connection.x1 = input.layer * 2 * width + width;
+                connection.y1 = 0;
+                connections.emplace_back(connection);
+            }
+        }
+    }
+
+
+}
+
+void setPosition(std::vector<Vertex>& elements, const int max_elements, const float height, const float width) {
+    if(elements.size() == max_elements) {
+        int count = 0;
+        for(Vertex& elem : elements) {
+            elem.position.x = elem.layer * HOR_SPACING * width;
+            elem.position.y = count * VER_SPACING * height;
+            elem.position.h = height;
+            elem.position.w = width;
+            count++;
+        }
+    }
+
+    else {
+        int count = 0;
+        int freeSpaces = elements.size() + 1;
+        float step = (SCREEN_HEIGHT - height * elements.size()) / freeSpaces;
+        for(Vertex& elem : elements) {
+            elem.position.x = elem.layer * HOR_SPACING * width;
+            elem.position.y = step * (count + 1) + height * count;
+            elem.position.h = height;
+            elem.position.w = width;
+            count++;
+        }
+    }
+}
+
+void setConnections(Vertex& element, const std::vector<std::vector<Vertex>> elements, std::vector<Line>& connections) {
+    for(auto input : element.ins) {
+        for(auto layered_elements : elements) {
+            for(auto compared_element : layered_elements) {
+                if(input.name == compared_element.name) {
+                    Line connect;
+                    connect.x1 = compared_element.position.x + compared_element.position.w;
+                    connect.y1 = compared_element.position.y + compared_element.position.h/2;
+                    connect.x2 = element.position.x;
+                    connect.y2 = element.position.y + element.position.h/2;
+                    connections.emplace_back(connect);
+                }
```

Comment:
`Line`-related calculations are better to be corresponding methods of the `Line` class.